modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -2548,6 +2552,9 @@ private String exchangeTimingsLogMessage(String header, List<String> timings) {
 
         final Throwable err0 = err;
 
+        if (err0 != null && cleanIdxRebuildFutures)
```

Comment:
this place is unclear for me, additional flag is always evel.

Reply:
I do not want to leave a possible memory leak that can occur due to the fact that we have passed the **init** stage but done with an error due to which it will not be possible to start rebuilding the indexes.